proof/refine/ARM/CNodeInv_R.thy

```diff
@@ -583,16 +583,36 @@ lemma suspend_ctes_of_thread:
   apply (case_tac cte, simp)
   done
 
+lemma schedContextUnbindTCB_ctes_of[wp]:
+  "\<lbrace>\<lambda>s. P (ctes_of s)\<rbrace>
+     schedContextUnbindTCB t
+   \<lbrace>\<lambda>_ s. P (ctes_of s)\<rbrace>"
+  apply (wpsimp simp: schedContextUnbindTCB_def wp: threadSet_ctes_ofT)
+     apply (clarsimp simp: ran_def tcb_cte_cases_def split: if_splits)
+  by wpsimp+
+
+crunches setConsumed, schedContextCompleteYieldTo, unbindNotification
+  for ctes_of[wp]: "\<lambda>s. P (ctes_of s)"
+  (simp: crunch_simps wp: crunch_wps)
+
+lemma schedContextUnbindTCB_ctes_of_thread:
```

Comment:
Do we need these three lemmas? It looks like the one above (`schedContextUnbindTCB_ctes_of`) would solve all of them.